pkg/iterator/emitters.go

```diff
@@ -76,7 +81,112 @@ func (e *TableEmitter) SnapshotMetadataIteratorRecord(recordNumber int, metadata
 		fmt.Fprintf(e.Writer, tableRowFmt, recordNumber, metadata.VolumeCapacityBytes, bmt, bmd.ByteOffset, bmd.SizeBytes)
 	}
 
-	return true
+	return nil
+}
+
+func (e *TableEmitter) SnapshotMetadataIteratorDone(_ int) error {
+	return nil
+}
+
+// VerifierEmitter will write the changed blocks from the source device to the target device at the designated offset and
+// compare the contents of the source and target devices at the end.
+type VerifierEmitter struct {
```

Comment:
There are no unit tests for this emitter. As this resides in this package it becomes part of the public interface and should be held to a higher standard.

Reply:
> There are no unit tests for this emitter. As this resides in this package it becomes part of the public interface and should be held to a higher standard.

I was working on it.
Its updated now.
Please take a look